File: EXAMPLES/py_lgpio/bench.py
#!/usr/bin/env python
"""
bench.py
2020-11-18
Public Domain

http://abyz.me.uk/lg/py_lgpio.html

./bench.py
"""
import time
import lgpio

OUT=21
LOOPS=100000

h = lgpio.gpiochip_open(0)

lgpio.gpio_claim_output(h, OUT)

t0 = time.time()

for i in range(LOOPS):
   lgpio.gpio_write(h, OUT, 0)
   lgpio.gpio_write(h, OUT, 1)

t1 = time.time()

lgpio.gpiochip_close(h)

print("{:.0f} toggles per second".format(LOOPS/(t1-t0)))

File: EXAMPLES/rgpio/bench.c
/*
bench.c
2020-11-18
Public Domain

http://abyz.me.uk/lg/rgpio.html

gcc Wall -o bench bench.c -lrgpio

./bench
*/

#include <stdio.h>
#include <stdlib.h>

#include <lgpio.h>
#include <rgpio.h>

#define LFLAGS 0

#define OUT 21
#define LOOPS 5000

int main(int argc, char *argv[])
{
   int sbc;
   int h;
   int i;
   double t0, t1;

   sbc = rgpiod_start(NULL, NULL);

   if (sbc < 0)
   {
      printf("connection failed\n");
      exit(-1);
   }

   h = gpiochip_open(sbc, 0);

   if (h >= 0)
   {
      if (gpio_claim_output(sbc, h, LFLAGS, OUT, 0) == LG_OKAY)
      {

         t0 = lgu_time();

         for (i=0; i<LOOPS; i++)
         {
            gpio_write(sbc, h, OUT, 0);
            gpio_write(sbc, h, OUT, 1);
         }

         t1 = lgu_time();

         printf("%.0f toggles per second\n", (1.0 * LOOPS)/(t1-t0));
      }

      gpiochip_close(sbc, h);
   }

   rgpiod_stop(sbc);
}

